src/backends/tensorrt/tensorrtlib.cc

```diff
@@ -884,28 +906,69 @@ namespace dd
                       break; // this belongs to next image
                     ++k;
                     outr += det_size;
+
                     if (detection[2] < output_params->confidence_threshold)
                       continue;
 
                     // Fix border of bboxes
-                    detection[3] = std::max(((float)detection[3]), 0.0f);
-                    detection[4] = std::max(((float)detection[4]), 0.0f);
-                    detection[5] = std::min(((float)detection[5]), 1.0f);
-                    detection[6] = std::min(((float)detection[6]), 1.0f);
+                    detection[3]
+                        = std::max(((float)detection[3]), 0.0f) * (cols - 1);
+                    detection[4]
+                        = std::max(((float)detection[4]), 0.0f) * (rows - 1);
+                    detection[5]
+                        = std::min(((float)detection[5]), 1.0f) * (cols - 1);
+                    detection[6]
+                        = std::min(((float)detection[6]), 1.0f) * (rows - 1);
 
                     probs.push_back(detection[2]);
                     cats.push_back(this->_mlmodel.get_hcorresp(detection[1]));
                     APIData ad_bbox;
-                    ad_bbox.add("xmin", static_cast<double>(detection[3]
-                                                            * (cols - 1)));
-                    ad_bbox.add("ymin", static_cast<double>(detection[4]
-                                                            * (rows - 1)));
-                    ad_bbox.add("xmax", static_cast<double>(detection[5]
-                                                            * (cols - 1)));
-                    ad_bbox.add("ymax", static_cast<double>(detection[6]
-                                                            * (rows - 1)));
+                    ad_bbox.add("xmin", static_cast<double>(detection[3]));
+                    ad_bbox.add("ymin", static_cast<double>(detection[4]));
+                    ad_bbox.add("xmax", static_cast<double>(detection[5]));
+                    ad_bbox.add("ymax", static_cast<double>(detection[6]));
                     bboxes.push_back(ad_bbox);
                   }
+
+                if (_need_nms)
+                  {
+                    std::vector<std::vector<double>> sorted_boxes;
```

Comment:
Maybe we could put the full nms processing as a function in `bbox_utils` instead of having it into `tensorrtlib` ?